app/code/Magento/InventorySales/Plugin/Sales/Shipment/CollectSourcesForShipmentItems.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Plugin\Sales\Shipment;
+
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\Order\ShipmentFactory;
+use Magento\InventoryShipping\Model\ShippingAlgorithmProviderInterface;
+
+/**
+ * This is the best entry point for both POST and API request
+ */
+class CollectSourcesForShipmentItems
+{
+    /**
+     * @var ShippingAlgorithmProviderInterface
+     */
+    private $shippingAlgorithmProvider;
+
+    /**
+     * ProcessShipmentItems constructor.
+     * @param ShippingAlgorithmProviderInterface $shippingAlgorithmProvider
+     */
+    public function __construct(
+        ShippingAlgorithmProviderInterface $shippingAlgorithmProvider
+    ) {
+        $this->shippingAlgorithmProvider = $shippingAlgorithmProvider;
+    }
+
+    /**
+     * @param ShipmentFactory $subject
+     * @param callable $proceed
+     * @param Order $order
+     * @param array $items
+     * @param null $tracks
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function aroundCreate(
```

Comment:
why do you need `aroundCreate ` but not `afterCreate` ? 
as you call process as the first line of your function
`$shipment = $proceed($order, $items, $tracks);`